### ISSUE TITLE
Step 2 is not running when accessing dataport via API

### ISSUE THREAD

**Issue by betterapp** [BlackbitDigitalCommerce/pimcore-data-director#75]
I have a dataport.

When I click on Start Complete Export button the result is:

![image](https://user-images.githubusercontent.com/7449124/215699309-56b571fe-f8f6-4d13-bed0-96ee1deeecf0.png)

And I can download file with data.

When I try to access this export via API call the result is:

![image](https://user-images.githubusercontent.com/7449124/215699575-6975418d-b853-45bb-8e68-a0aeef9d44e0.png)

Second step is not running and I get empty file.

Why ?

DD: 3.2.35

**Comment by BlackbitDevs** [comment]
This happens when the cache is being used. The response documents get cached on the server. When export recognizes that since last execution with the given input parameters, nothing has changed in the data, then the result from the cache gets delivered and step 2 of the export gets completely skipped.
When the export gets executed manually, it gets executed with `force=1`. In this case the cache does not get used. This is why you see different behaviour.
The question now is, why you get an empty file. 

For now, could you please remove all files in `var/cache/BlackbitDataDirector/`? Afterwards it should work.
To completely bypass the caching behaviour you can provide `force=1` in your REST API request - but actually it would be better for performance reasons if it worked without this. So if this happens again, even if you have cleaned all the cache files in `var/cache/BlackbitDataDirector/`, please notify me again here.

**Comment by betterapp** [comment]
Yes. After clearing cache it is working. But why I needed to clear the cache ?

**Comment by BlackbitDevs** [comment]
When you got an empty response document, the reason was that in a previous export this empty response document got cached. So maybe there is a problem with cache pruning but you are the first who notifies me about this. To eliminate the possible cause that a debug statement in one of your callback functions was responsible, it was necessary to clear the cache. If the problem does not occur again, everything is fine. If it does occur again, please notify me, then I will take a closer look.
